core/include/msg.h

```diff
@@ -356,13 +356,32 @@ int msg_reply(msg_t *m, msg_t *reply);
 int msg_reply_int(msg_t *m, msg_t *reply);
 
 /**
- * @brief Check how many messages are available in the message queue
+ * @brief Check how many messages are available (waiting) in the message queue
+ *        of a specific thread
+ *
+ * @param[in] pid    a PID
+ *
+ * @return Number of messages available in queue of @p pid on success
+ * @return -1, if no caller's message queue is initialized
+ */
+int msg_avail_thread(kernel_pid_t pid);
+
+/**
+ * @brief Check how many messages are available (waiting) in the message queue
  *
  * @return Number of messages available in our queue on success
  * @return -1, if no caller's message queue is initialized
  */
 int msg_avail(void);
 
+/**
+ * @brief Get maximum capacity of a thread's queue length
+ *
+ * @return Number of total messages that fit in the queue of @p pid on success
+ * @return -1, if no caller's message queue is initialized
+ */
+int msg_queue_capacity(kernel_pid_t pid);
```

Comment:
Maybe rename to `msg_queue_capacity_thread()` for naming consistency? And add 

```C
static inline int msg_queue_capacity(void)
{
    retrurn msg_queue_capacity_thread(thread_getpid());
}
```

Even if there is no obvious use case for this now, it is hard to change APIs later on.

Reply:
This causes a cyclic inclusion, as `thread.h` (where `thread_getpid()` is declared) is including `msg.h`. I would have used `sched_active_pid`, but that seems not to be exposed anymore. Last alternative would be to make it a non-`static inline` function.